### ISSUE TITLE
Cached node private key causes error when switching to production endpoint

### ISSUE THREAD

**Issue by openfirmware** [schubergphilis/letsencrypt#29]
First of all, thanks for the cookbook. It works great!

I ran into a problem yesterday when testing a node with multiple new LE certificates while using the staging endpoint for ACME. After I was able to get those certificates working, I switched the ACME endpoint in the node attributes to the production endpoint. Then the chef client run failed with the error:

    Acme::Client::Error::Unauthorized: No registration exists matching provided key

It took me a few hours to figure out but when the chef client ran with the staging endpoint, a private key was stored in the node attributes. Then switching to the production endpoint would mean chef would try to register LE certificates using the staging endpoint's private key, which doesn't work with the production endpoint. (For clarification, this private key is not the same as the certificate private key, it is the private key for communication with the ACME endpoint.)

I fixed it by removing the `node['letsencrypt']['private_key']` attribute from the node — I am using Chef Zero, so editing `nodes/<nodename>.json` was enough to clear it. For Chef Server users I would think using `$ knife node edit` would work.

This isn't really a bug, but perhaps it would be helpful to note this either here or in the README for anyone else who runs into the problem.

**Comment by thoutenbos** [comment]
Default endpoint set to production now (in #30)

**Comment by notapatch** [comment]
You see this when you switch between endpoints (though I have only observed it when moving from staging to production). There is no obvious fix for this - just to be aware of the issue.

I deleted the private_key in the script when changing endpoints ([there is a bug in Chef to stop you setting an attribute to nil](https://github.com/chef/chef/issues/1507) if it has already been set).

    node.rm('acme', 'private_key')

Everything works as normal then - thanks for the cookbook.

[1]: https://github.com/chef/chef/issues/1507
